client/asset/dcr/rpcwallet.go

```diff
@@ -138,44 +143,46 @@ func newRPCWallet(cfg *Config, chainParams *chaincfg.Params, logger dex.Logger)
 		log:         log,
 	}
 
+	certs, err := os.ReadFile(cfg.RPCCert)
+	if err != nil {
+		return nil, fmt.Errorf("TLS certificate read error: %w", err)
+	}
+
 	log.Infof("Setting up rpc client to communicate with dcrwallet at %s with TLS certificate %q.",
 		cfg.RPCListen, cfg.RPCCert)
-	err := rpcw.setupRPCClient(cfg.RPCListen, cfg.RPCUser, cfg.RPCPass, cfg.RPCCert)
+	rpcw.rpcCfg = &rpcclient.ConnConfig{
+		Host:                cfg.RPCListen,
+		Endpoint:            "ws",
+		User:                cfg.RPCUser,
+		Pass:                cfg.RPCPass,
+		Certificates:        certs,
+		DisableConnectOnNew: true,
+	}
+	err = rpcw.setupRPCClient() // this rpcWallet instance is still locally scoped, no need to lock rpcMtx
```

Comment:
Because we have `DisableConnectOnNew` set, there's probably no way we're going to error in `setupRPCClient`, so doing it in the constructor is probably pointless in terms of checking configuration. Does it simplify things to defer all calls of `setupRPCClient` to `Connect`?